### PR TITLE
Add early test to check router pods are not on same node

### PR DESCRIPTION
**Pull request by xueqzhan**

### CODE REVIEW

File: test/extended/router/pods.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+	g.BeforeEach(func() {
+		var err error
+		_, err = exutil.WaitForRouterServiceIP(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.Describe("The HAProxy router pods", func() {
+		g.It("should be scheduled on different nodes", func() {
+			pods, err := oc.KubeFramework().ClientSet.CoreV1().Pods("openshift-ingress").List(context.Background(), metav1.ListOptions{})
```
Comment (deads2k) [inline test/extended/router/pods.go:27]: Please abstract this into a 

```go
// requirePodsOnDifferentNodesTest fails if any pods with the prefix in the namespace are scheduled to the same node
type requirePodsOnDifferentNodesTest struct{
    namespace string
    podPrefix string
}

func (p requirePodsonDifferentNodesTest) run(whatever args){
    pods, err := oc.KubeFramework().ClientSet.CoreV1().Pods("openshift-ingress").List(context.Background(), metav1.ListOptions{})
    // until the end
}

g.Describe("The HAProxy router pods", func() {
    g.It("should be scheduled on different nodes", func() {
        requirePodsOnDifferentNodesTest{namespace:, podPrefix}.run(args)
    }
}

g.Describe("The openshift-apiserver pods", func() {
    g.It("should be scheduled on different nodes", func() {
        requirePodsOnDifferentNodesTest{namespace: openshift-apiserver, podPrefix: openshift-apiserver-}.run(args)
    }
}

g.Describe("The oauth-apiserver pods", func() {
    g.It("should be scheduled on different nodes", func() {
        requirePodsOnDifferentNodesTest{namespace: oauth-apiserver, podPrefix: oauth-apiserver-}.run(args)
    }
}


```

File: test/extended/router/pods.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	"strings"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = g.Describe("[sig-network][Early][Feature:Router]", func() {
```
Comment (deads2k) [inline test/extended/router/pods.go:14]: sig-scheduling

Comment (deads2k) [inline test/extended/router/pods.go:14]: not [Feature:Router] this is checking scheduling

**Comment by deads2k** [comment]
need to run hack/update-generated something

**Comment by deads2k** [comment]
/approve
/override ci/prow/e2e-gcp 
/override ci/prow/e2e-gcp-upgrade 

override because of image pull on b02

**Comment by deads2k** [comment]
/assign @stbenjam 

if this appears to be working, then we'll need to make an [early] invariant check before we run any upgrade tests since upgrades will clear this problem out.

**Comment by deads2k** [comment]
we'll start here and likely have to make this execute when the test first begins.  Before starting the ugprade.

/lgtm

**Comment by omertuc** [comment]
This is consistently failing on single-node clusters. Please revert / fix this

**Comment by omertuc** [comment]
Okay fix seems simple, no need to revert, created #26905

**Pull request merged**